### PR TITLE
11 get student

### LINKED ISSUE #11: Get Student
**Trigger:**

After the QR code is scanned
Clicked View student
After the student is registered by the cashier

**Page:**
Image
Image
Image

**Requirements:**

- [x] The endpoint should be '/student/{uuid}'
- [x] Return the student's information

### PR DESCRIPTION
**Pull request by louispawaon**
- Working /student/{uuid} endpoint
- Should return the following:

```
{
        uuid
        firstName
        lastName
        email
        year_and_course
}
```
- A not found student returns:
```
{ Student not found }
```
- Added .env.example for other devs to be guided what to put inside their local .env files
- Added type for url in prisma.service.ts to prevent type errors

### CODE REVIEW

File: src/student/student.service.ts [hunk 1]
```diff
@@ -19,15 +20,30 @@ export class StudentService {
...
-
   update(id: number, updateStudentDto: UpdateStudentDto) {
     return `This action updates a #${id} student`;
   }
 
   remove(id: number) {
     return `This action removes a #${id} student`;
   }
+
+  async getStudentByUuid(uuid: string): Promise<ReadStudentDto> {
```
Comment (rhaeyx) [inline src/student/student.service.ts:31]: hi ya, pwede pa apil ug return sa data for event and sa payment na connected sa student. thank you!

**Review by rhaeyx — CHANGES_REQUESTED** [review]

File: src/student/student.service.ts (hunk 1 above)
Comment (louispawaon) [reply src/student/student.service.ts:31]: return lang nako ang eventid ug paymentid or do u want na ang unod sa event og payment pod ang ireturn?

Comment (rhaeyx) [reply src/student/student.service.ts:31]: apila lang ang content mismo ya

**Comment by louispawaon** [comment]
> * Working /student/{uuid} endpoint
> * Should return the following:
> 
> ```
> {
>         uuid
>         firstName
>         lastName
>         email
>         year_and_course
> }
> ```
> 
> * A not found student returns:
> 
> ```
> { Student not found }
> ```
> 
> * Added .env.example for other devs to be guided what to put inside their local .env files
> * Added type for url in prisma.service.ts to prevent type errors

should now return the following:

```
{
        uuid
        firstName
        lastName
        email
        year_and_course
        event: {
            createdAt
            updatedAt
            title
            price
            requires_payment
            max_participants
            description
            date
            is_active
            form_name
          };
          payment: {
            createdAt
            updatedAt
            photo_src
            status
          };
}
```

**Review by rhaeyx — APPROVED** [review]

**Comment by angellialkuino** [comment]
Shortens code, returns all the same needed info
```
this.prisma.student.findUnique({
      where: { uuid },
      include: { event: true, payment: true }
})
```

**Pull request merged**